### PR TITLE
fix: cache actions for PR jobs

### PR DESCRIPTION
**Pull request by d2lam**
Related: https://github.com/screwdriver-cd/screwdriver/issues/1382

### CODE REVIEW

File: store-cli.go [hunk 1]
```diff
@@ -39,10 +39,26 @@ func finalRecover() {
 	successExit()
 }
 
+// Skip cache action for PR jobs (event, pipeline scope)
+func skipCache(storeType, scope, action string) bool {
+	// if is not cache, or if job is not PR
+	if (storeType != "cache" || os.Getenv("SD_PULL_REQUEST") == "") {
+		return false
+	}
```
Comment (pranavrc) [inline store-cli.go:48]: Do we need this block? Can just add ` if (storeType === "cache" && <IS_PULL_REQUEST>)` to the condition below

File: store-cli.go [hunk 2]
```diff
@@ -82,6 +98,11 @@ func makeURL(storeType, scope, key string) (*url.URL, error) {
 }
 
 func get(storeType, scope, key string) error {
+	skip := skipCache(storeType, scope, "get")
+	if skip == true {
```
Comment (pranavrc) [inline store-cli.go]: `if skip {`?

File: store-cli.go [hunk 3]
```diff
@@ -104,6 +125,10 @@ func get(storeType, scope, key string) error {
 }
 
 func set(storeType, scope, filePath string) error {
+	skip := skipCache(storeType, scope, "set")
+	if skip == true {
```
Comment (pranavrc) [inline store-cli.go]: `if skip {`?

File: store-cli.go [hunk 4]
```diff
@@ -124,6 +149,11 @@ func set(storeType, scope, filePath string) error {
 }
 
 func remove(storeType, scope, key string) error {
+	skip := skipCache(storeType, scope, "remove")
+	if skip == true {
```
Comment (pranavrc) [inline store-cli.go]: `if skip {`?

File: store-cli.go [hunk 5]
```diff
@@ -104,6 +125,10 @@ func get(storeType, scope, key string) error {
 }
 
 func set(storeType, scope, filePath string) error {
+	skip := skipCache(storeType, scope, "set")
+	if skip == true {
+		return nil
```
Comment (pranavrc) [inline store-cli.go:130]: Can probably log something like "Pull request cannot set or remove pipeline cache, skipping` here

File: store-cli.go (hunk 1 above)
Comment (d2lam) [reply store-cli.go:48]: The condition below returns `true`, this one is short circuit for `false` case

File: store-cli.go (hunk 2 above)
Comment (d2lam) [reply store-cli.go]: fixed. Thought you need to be explicit in go

File: store-cli.go (hunk 1 above)
Comment (pranavrc) [reply store-cli.go:48]: What I meant was, just have one condition `if (<IS_PULL_REQUEST> && storeType === "cache" && scope === "pipeline" && action != "get") { return true }`, but this should be ok

**Review by pranavrc — APPROVED** [review]
image

**Pull request merged**